src/clr-angular/forms/radio/radio-container.ts

```diff
@@ -4,23 +4,67 @@
  * The full license information can be found in LICENSE in the root directory of this project.
  */
 
-import { Component } from '@angular/core';
+import { Component, ContentChild, OnDestroy, Optional } from '@angular/core';
+import { Subscription } from 'rxjs';
 
 import { DynamicWrapper } from '../../utils/host-wrapping/dynamic-wrapper';
-import { ControlIdService } from '../common/providers/control-id.service';
+import { IfErrorService } from '../common/if-error/if-error.service';
+import { ClrLabel } from '../common/label';
+import { ControlClassService } from '../common/providers/control-class.service';
+import { LayoutService } from '../common/providers/layout.service';
+import { NgControlService } from '../common/providers/ng-control.service';
 
 @Component({
   selector: 'clr-radio-container',
   template: `
-        <!-- We want the radio input to be before the label, always -->
-        <ng-content select="[clrRadio]"></ng-content>
-        <ng-content></ng-content>
-        <label *ngIf="_dynamic"></label>
+    <ng-content select="label"></ng-content>
+    <label *ngIf="!label && addGrid()"></label>
+    <div class="clr-control-container" [ngClass]="controlClass()">
+      <ng-content select="clr-radio-wrapper"></ng-content>
+      <ng-content select="clr-control-helper" *ngIf="!invalid"></ng-content>
+      <clr-icon *ngIf="invalid" class="clr-validate-icon" shape="exclamation-circle"></clr-icon>
+      <ng-content select="clr-control-error" *ngIf="invalid"></ng-content>
+    </div>
     `,
-  host: { '[class.radio]': 'true' },
-  providers: [ControlIdService],
+  host: {
+    '[class.clr-form-control]': 'true',
+    '[class.clr-row]': 'addGrid()',
+  },
+  providers: [NgControlService, ControlClassService, IfErrorService, NgControlService],
 })
-export class ClrRadioContainer implements DynamicWrapper {
-  // Indicates whether the container is dynamically created by the radio button itself
+export class ClrRadioContainer implements DynamicWrapper, OnDestroy {
+  subscriptions: Subscription[] = [];
   _dynamic = false;
+  invalid = false;
+  @ContentChild(ClrLabel) label: ClrLabel;
+  controls = [];
+
+  constructor(
+    private ifErrorService: IfErrorService,
+    @Optional() private layoutService: LayoutService,
+    private controlClassService: ControlClassService
+  ) {
+    this.subscriptions.push(
+      this.ifErrorService.statusChanges.subscribe(control => {
+        this.invalid = control.invalid;
+      })
+    );
+  }
+
+  controlClass() {
```

Comment:
The same comment as in the `clrSelect` PR. This method is just calling another method and has no dependency injections. It could be a simple getter property.